joomla/joomla-cms
# [6.2] Rework how URLs are injected into applications

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/38518 and https://github.com/joomla/joomla-cms/pull/38524 .

I didn't like the fix that was merged. But didn't have time to do anything other than moan in the maintainers internal chat during the week. Finally had some time to sit down and test out options during the weekend. This is a likely better architectural piece of code (as we're not going to setup custom super globals - the application controls it all) - but comes with more complexity. I don't *believe* it breaks any b/c (if it's going to be anywhere it's around me having to slightly move where we set the app root - it has to be outside the constructor else we hit a circular loop - but it's before any extensions are loaded so I think it's fine) but definitely needs some serious testing.

### Summary of Changes
Injects the main request Uri from the application (where we already had the duplicate code)

### Testing Instructions
Check all 5 applications (console/old style cli/administrator/site/api) work as before patch. Check URL is correctly deduced and the original issue within #38518 is still fixed after applying this patch.

### Documentation Changes Required
None?

//cc @nikosdion @HLeithner @roland-d @fancyFranci for careful review
